Mathlib/GroupTheory/SpecificGroups/Cyclic.lean

```diff
@@ -131,14 +131,20 @@ theorem isCyclic_of_orderOf_eq_card [Fintype α] (x : α) (hx : orderOf x = Fint
 @[deprecated] -- 2024-02-21
 alias isAddCyclic_of_orderOf_eq_card := isAddCyclic_of_addOrderOf_eq_card
 
+@[to_additive]
+theorem Subgroup.eq_bot_or_eq_top_of_prime_card {G : Type*} [Group G] {_ : Fintype G}
+    (H : Subgroup G) {p : ℕ} [hp : Fact p.Prime] (h : Fintype.card G = p) : H = ⊥ ∨ H = ⊤ := by
+  classical
+  have := card_subgroup_dvd_card H
+  rwa [h, Nat.dvd_prime hp.1, ← h, ← Nat.card_eq_fintype_card, ← Nat.card_eq_fintype_card,
+    ← eq_bot_iff_card, card_eq_iff_eq_top] at this
+
 /-- Any non-identity element of a finite group of prime order generates the group. -/
 @[to_additive "Any non-identity element of a finite group of prime order generates the group."]
 theorem zpowers_eq_top_of_prime_card {G : Type*} [Group G] {_ : Fintype G} {p : ℕ}
     [hp : Fact p.Prime] (h : Fintype.card G = p) {g : G} (hg : g ≠ 1) : zpowers g = ⊤ := by
```

Comment:
What do you think about assuming `Fact (Fintype.card G).Prime` and get rid of `p`?
```suggestion
theorem Subgroup.eq_bot_or_eq_top_of_prime_card {G : Type*} [Group G] {_ : Fintype G}
    (H : Subgroup G) {p : ℕ} [hp : Fact p.Prime] (h : Fintype.card G = p) : H = ⊥ ∨ H = ⊤ := by
  classical
  have := card_subgroup_dvd_card H
  rwa [h, Nat.dvd_prime hp.1, ← h, ← Nat.card_eq_fintype_card, ← Nat.card_eq_fintype_card,
    ← eq_bot_iff_card, card_eq_iff_eq_top] at this

/-- Any non-identity element of a finite group of prime order generates the group. -/
@[to_additive "Any non-identity element of a finite group of prime order generates the group."]
theorem zpowers_eq_top_of_prime_card {G : Type*} [Group G] {_ : Fintype G} {p : ℕ}
    [hp : Fact p.Prime] (h : Fintype.card G = p) {g : G} (hg : g ≠ 1) : zpowers g = ⊤ := by
```

Reply:
Good point. I've made this change in the new helper lemma `eq_bot_or_eq_top_of_prime_card`, but making this change throughout the whole file should probably wait for another PR (maybe the one where I switch this file over to `Nat.card`).